packages/devtools_app/lib/src/screens/debugger/debugger_screen.dart

```diff
@@ -447,20 +447,21 @@ class _FloatingDebuggerControlsState extends State<FloatingDebuggerControls>
     with
         AutoDisposeMixin,
         ProvidedControllerMixin<DebuggerController, FloatingDebuggerControls> {
-  late bool paused;
-
   late double controlHeight;
 
+  bool get _isPaused => serviceManager.isMainIsolatePaused;
+
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     if (!initController()) return;
-    paused = serviceManager.appState.isPaused.value;
-    controlHeight = paused ? defaultButtonHeight : 0.0;
-    addAutoDisposeListener(serviceManager.appState.isPaused, () {
+    cancelListeners();
+
+    controlHeight = _isPaused ? defaultButtonHeight : 0.0;
+    addAutoDisposeListener(
+        serviceManager.isolateManager.mainIsolateState!.isPaused, () {
```

Comment:
could isMainIsolatePaused just be a notifier too?
I feel like there is a bit of a disconnect by using isMainIsolatePaused as a proxy to state inside the isolateManager, and then needing to know that the two are connected if you ever want to monitor that the state has changed.
also will _isPaused update here, if the paused state changes?

Reply:
isMainIsolatePaused is a helper method. It's value is shorten the code. If I make it notifier, the code will increase by `.value` in many places. 
Yes, there are pros and cons here. But, whoever would want to listen to the value, will start with reading the method body and will understand how to monitor the state.
Added documentation to decrease confusion.